### PR TITLE
[P2P] Throttle peer discovery

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/PeerSelector.cs [hunk 1]
```diff
@@ -187,9 +191,9 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
         /// <inheritdoc/>
         public IEnumerable<PeerAddress> SelectPeersForDiscovery(int peerCount)
         {
-            // Randomly order the list of peers and return the amount asked for.
-            var allPeers = this.peerAddresses.OrderBy(p => this.random.Next());
-            return allPeers.Select(p => p.Value).Take(1000);
+            var discoverable = this.peerAddresses.Values.Where(p => p.LastDiscoveredFrom < DateTimeProvider.Default.GetUtcNow().AddHours(-PeerSelector.DiscoveryThresholdHours));
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/PeerSelector.cs]: You should probably use here the injected property `DateTimeProvider.Default`

Comment (fassadlr) [reply src/Stratis.Bitcoin/P2P/PeerSelector.cs]: Fixed 👍

File: src/Stratis.Bitcoin/P2P/PeerSelector.cs [hunk 2]
```diff
@@ -187,9 +195,9 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
         /// <inheritdoc/>
         public IEnumerable<PeerAddress> SelectPeersForDiscovery(int peerCount)
         {
-            // Randomly order the list of peers and return the amount asked for.
-            var allPeers = this.peerAddresses.OrderBy(p => this.random.Next());
-            return allPeers.Select(p => p.Value).Take(1000);
+            var discoverable = this.peerAddresses.Values.Where(p => p.LastDiscoveredFrom < this.dateTimeProvider.GetUtcNow().AddHours(-PeerSelector.DiscoveryThresholdHours));
+            var allPeers = discoverable.Take(1000).ToList();
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/PeerSelector.cs]: Should we not first apply randomness and then take to top 1000?

Comment (fassadlr) [reply src/Stratis.Bitcoin/P2P/PeerSelector.cs]: @dangershony Ah yes you right. Fixed...

**Pull request merged**